server/log_rpc_server.go

```diff
@@ -508,7 +508,11 @@ func (t *TrillianLogRPCServer) GetEntryAndProof(ctx context.Context, req *trilli
 	}
 	r := &trillian.GetEntryAndProofResponse{SignedLogRoot: &root}
 
-	// FIXME: should we return a proof to the available tree size if req.LeafIndex < root.TreeSize?
+	if req.TreeSize > root.TreeSize && req.LeafIndex < root.TreeSize {
```

Comment:
Ah, this does not do what I understood the `FIXME` comment to mean!
> FIXME: should we return a proof to the available tree size if req.LeafIndex < root.TreeSize?

I understood that to say if the leaf is within *our* [implicitly larger] tree, then return a proof to that rather than the treeSize the client asked for.

But the this change is not that and is ok :)